### DISCUSSION TITLE
Multiple connection to DB, Alembic Hangs

### DISCUSSION THREAD

**Discussion by KhadijaMahanga** [sqlalchemy/alembic#1097] [Usage Questions]
Every time I run upgrade I have to first terminate all connections to my PostgreSQL database. Otherwise it hangs.

What should I do to make sure I don't need to terminate other connection?

**Comment by zzzeek** [comment]
quite simply don't use multiple connections in an alembic migration.  use a single connection / transaction for the entire operation, so that DDL operations aren't deadlocking against other operations.

**Reply by KhadijaMahanga** [reply]
How can I do that? I don't see where I am using multiple connections in my migrations.

**Reply by zzzeek** [reply]
OK, you named this discussion "multiple connection to DB", so it seemed you were intending to use multiple connections.

Alembic's default behavior is to use just a single connection.   

When you say "multiple connections", do you mean, you have other processes connecting to your PG database at the same time? *those* connections?      if you have connections to your database that are idle in transactions, and alembic is attempting to affect those same tables, that would hang also, sure.    it's not clear what the specifics of this "hang" are.    take a look at [pg_stat_activity](https://www.depesz.com/2022/07/05/understanding-pg_stat_activity/) - those connections that have affected tables you want to migrate, need to be released from an idle transaction.  that's just how PostgreSQL works.

**Reply by KhadijaMahanga** [reply]
Yes, they are other process connecting to my PG DB. Thank you, I will look at pg_stat_activity
